File: 프로그래머스/3/43238. 입국심사/입국심사.py
def solution(n, times):
    answer = 0
    left, right = 0, max(times)*n
    while left<=right:
        mid = (left+right)//2
        count = sum(mid//t for t in times)
        if count>=n:
            answer = mid
            right = mid-1
        else:
            left = mid+1
    return answer

File: 프로그래머스/3/43238. 입국심사/입국심사.java
class Solution {
    public long solution(int n, int[] times) {
        long left = 0;
        
        long right = 0;
        for (int time: times) {
            right = Math.max(right, (long) time);
        }
        right *= n;
        
        long mid = (left + right)/2;
        
        while (left <= right) {
            mid = (left + right)/2;
            
            long count = 0;
            for (int time: times) {
                count += mid/time;
            }
            if (count >= n){
                right = mid-1;
            } else {
                left = mid + 1;
            }
        }
        return left;
    }
}